NetProbe/Utility/TestStorage.mm

```diff
@@ -0,0 +1,120 @@
+//
+//  TestStorage.m
+//  NetProbe
+//
+//  Created by Lorenzo Primiterra on 11/06/16.
+//  Copyright © 2016 Simone Basso. All rights reserved.
+//
+
+#import "TestStorage.h"
+
+@implementation TestStorage
+
++ (NSArray*)get_tests{
+    if ([[NSUserDefaults standardUserDefaults] objectForKey:@"tests"]){
+        return [[NSUserDefaults standardUserDefaults] objectForKey:@"tests"];
+    }
+    return nil;
+}
+
++ (void)add_test:(NetworkMeasurement*)test{
+    [self checkTests];
+    NSMutableArray *cache = [[[NSUserDefaults standardUserDefaults] arrayForKey:@"tests"] mutableCopy];
+    [cache addObject:[NSKeyedArchiver archivedDataWithRootObject:test]];
+    [[NSUserDefaults standardUserDefaults] setObject:cache forKey:@"tests"];
+    [[NSUserDefaults standardUserDefaults] synchronize];
+}
+
++ (NSArray*)remove_test:(NSNumber*)test_id{
+    [self checkTests];
+    NSMutableArray *cache = [[[NSUserDefaults standardUserDefaults] arrayForKey:@"tests"] mutableCopy];
+    for (int i = 0; i < [cache count]; i++) {
+        NetworkMeasurement* test = [NSKeyedUnarchiver unarchiveObjectWithData:[cache objectAtIndex:i]];
+        if (test.test_id == test_id){
+            [self removeFile:test.json_file];
+            [self removeFile:test.log_file];
+            [cache removeObjectAtIndex:i];
```

Comment:
Here I think you should `break` since you are looking for just one object inside the cache, or am I missing something obvious? If instead you want to continue iterating over the array, wouldn't you skip one element because you remove the element at `i`, then element `i+1` becomes `i` but you next iteration (if any) is on `i+1` (i.e. `i+2` before the element was removed?